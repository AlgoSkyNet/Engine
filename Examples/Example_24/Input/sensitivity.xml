<SensitivityAnalysis>
  <DiscountCurves>
    <DiscountCurve ccy="EUR">
      <ShiftType>Absolute</ShiftType>
      <ShiftSize>0.0001</ShiftSize>
      <ShiftTenors>6M,1Y,2Y,3Y,5Y,7Y,10Y,15Y,20Y</ShiftTenors>
    </DiscountCurve>
    <DiscountCurve ccy="USD">
      <ShiftType>Absolute</ShiftType>
      <ShiftSize>0.0001</ShiftSize>
      <ShiftTenors>6M,1Y,2Y,3Y,5Y,7Y,10Y,15Y,20Y</ShiftTenors>
    </DiscountCurve>
  </DiscountCurves>
  <IndexCurves>
    <IndexCurve index="EUR-EURIBOR-6M">
      <ShiftType>Absolute</ShiftType>
      <ShiftSize>0.0001</ShiftSize>
      <ShiftTenors>6M,1Y,2Y,3Y,5Y,7Y,10Y,15Y,20Y</ShiftTenors>
    </IndexCurve>
  </IndexCurves>
  <YieldCurves>
    <YieldCurve name="Lufthansa">
      <CurveType>EquityForecast</CurveType>
      <ShiftType>Absolute</ShiftType>
      <ShiftSize>0.0001</ShiftSize>
      <ShiftTenors>6M,1Y,2Y,3Y,4Y,5Y,7Y,10Y</ShiftTenors>
    </YieldCurve>
  </YieldCurves>
  <FxSpots>
    <FxSpot ccypair="USDEUR">
      <ShiftType>Relative</ShiftType>
      <ShiftSize>0.01</ShiftSize>
    </FxSpot>
  </FxSpots>
  <EquitySpots>
    <EquitySpot equity="Lufthansa">
      <ShiftType>Relative</ShiftType>
      <ShiftSize>0.01</ShiftSize>
    </EquitySpot>
  </EquitySpots>
  <EquityVolatilities>
    <EquityVolatility equity="Lufthansa">
      <ShiftType>Relative</ShiftType>
      <ShiftSize>0.01</ShiftSize>
      <ShiftExpiries>6M,1Y,2Y,3Y,5Y</ShiftExpiries>
      <ShiftStrikes/>
    </EquityVolatility>
  </EquityVolatilities>
  <CommodityCurves>
    <CommodityCurve name="COMDTY_GOLD_USD">
      <Currency>USD</Currency>
      <ShiftType>Relative</ShiftType>
      <ShiftSize>0.01</ShiftSize>
      <ShiftTenors>0D,6M,1Y,2Y,5Y,10Y</ShiftTenors>
    </CommodityCurve>
    <CommodityCurve name="COMDTY_WTI_USD">
      <Currency>USD</Currency>
      <ShiftType>Relative</ShiftType>
      <ShiftSize>0.01</ShiftSize>
      <ShiftTenors>0D,6M,1Y,2Y,5Y,10Y</ShiftTenors>
    </CommodityCurve>
  </CommodityCurves>
  <CrossGammaFilter>
    <Pair>DiscountCurve/EUR,DiscountCurve/EUR</Pair>
    <Pair>IndexCurve/EUR,IndexCurve/EUR</Pair>
    <Pair>DiscountCurve/EUR,IndexCurve/EUR</Pair>
  </CrossGammaFilter>
</SensitivityAnalysis>
