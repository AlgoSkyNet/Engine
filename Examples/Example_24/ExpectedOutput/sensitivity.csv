#TradeId,IsPar,Factor_1,ShiftSize_1,Factor_2,ShiftSize_2,Currency,Base NPV,Delta,Gamma
CommodityForward_Gold_1Y,false,DiscountCurve/USD/1/1Y,0.000000,,0.000000,USD,1398.71,-0.14,0.00
CommodityForward_Gold_1Y,false,CommodityCurve/COMDTY_GOLD_USD/0/1Y,0.000000,,0.000000,USD,1398.71,-11540.27,0.00
CommodityForward_Gold_Apr_17,false,DiscountCurve/USD/1/1Y,0.000000,,0.000000,USD,-8.90,0.00,0.00
CommodityForward_Gold_Apr_17,false,DiscountCurve/USD/2/2Y,0.000000,,0.000000,USD,-8.90,0.00,0.00
CommodityForward_Gold_Apr_17,false,CommodityCurve/COMDTY_GOLD_USD/0/1Y,0.000000,,0.000000,USD,-8.90,5441.78,0.00
CommodityForward_Gold_Apr_17,false,CommodityCurve/COMDTY_GOLD_USD/1/5Y,0.000000,,0.000000,USD,-8.90,324.80,0.00
CommodityForward_WTI_2Y,false,DiscountCurve/USD/2/2Y,0.000000,,0.000000,USD,-1669620.66,334.35,-0.07
CommodityForward_WTI_2Y,false,CommodityCurve/COMDTY_WTI_USD/0/1Y,0.000000,,0.000000,USD,-1669620.66,328104.54,0.00
CommodityForward_WTI_2Y,false,CommodityCurve/COMDTY_WTI_USD/1/5Y,0.000000,,0.000000,USD,-1669620.66,109268.39,0.00
CommodityForward_WTI_Oct_21,false,DiscountCurve/USD/4/5Y,0.000000,,0.000000,USD,81241.18,-25.73,0.01
CommodityForward_WTI_Oct_21,false,DiscountCurve/USD/5/7Y,0.000000,,0.000000,USD,81241.18,-20.89,0.01
CommodityForward_WTI_Oct_21,false,CommodityCurve/COMDTY_WTI_USD/1/5Y,0.000000,,0.000000,USD,81241.18,-197727.68,0.00
CommodityForward_WTI_Oct_21,false,CommodityCurve/COMDTY_WTI_USD/2/10Y,0.000000,,0.000000,USD,81241.18,-36081.62,0.00
CommodityOption_Gold_Apr_17,false,DiscountCurve/USD/1/1Y,0.000000,,0.000000,USD,-23540.49,1.83,0.00
CommodityOption_Gold_Apr_17,false,DiscountCurve/USD/2/2Y,0.000000,,0.000000,USD,-23540.49,1.06,0.00
CommodityOption_Gold_Apr_17,false,CommodityCurve/COMDTY_GOLD_USD/0/1Y,0.000000,,0.000000,USD,-23540.49,2510.75,-199.77
CommodityOption_Gold_Apr_17,false,CommodityCurve/COMDTY_GOLD_USD/1/5Y,0.000000,,0.000000,USD,-23540.49,155.43,-0.71
CommodityOption_Gold_Apr_17,false,CommodityVolatility/COMDTY_GOLD_USD/1/1Y/ATM,0.000000,,0.000000,USD,-23540.49,-172.32,-0.46
CommodityOption_Gold_Apr_17,false,CommodityVolatility/COMDTY_GOLD_USD/5/5Y/ATM,0.000000,,0.000000,USD,-23540.49,-63.29,-0.46
CommodityOption_WTI_Oct_21,false,DiscountCurve/USD/4/5Y,0.000000,,0.000000,USD,2355680.84,-746.13,0.24
CommodityOption_WTI_Oct_21,false,DiscountCurve/USD/5/7Y,0.000000,,0.000000,USD,2355680.84,-605.77,0.16
CommodityOption_WTI_Oct_21,false,CommodityCurve/COMDTY_WTI_USD/1/5Y,0.000000,,0.000000,USD,2355680.84,109204.89,2576.95
CommodityOption_WTI_Oct_21,false,CommodityCurve/COMDTY_WTI_USD/2/10Y,0.000000,,0.000000,USD,2355680.84,19736.57,85.82
CommodityOption_WTI_Oct_21,false,CommodityVolatility/COMDTY_WTI_USD/37/5Y/1.5,0.000000,,0.000000,USD,2355680.84,5906.31,44.07
CommodityOption_WTI_Oct_21,false,CommodityVolatility/COMDTY_WTI_USD/39/10Y/1.5,0.000000,,0.000000,USD,2355680.84,1668.95,15.44
CommodityOption_WTI_Oct_21,false,CommodityVolatility/COMDTY_WTI_USD/45/5Y/1.667,0.000000,,0.000000,USD,2355680.84,12698.94,58.23
CommodityOption_WTI_Oct_21,false,CommodityVolatility/COMDTY_WTI_USD/47/10Y/1.667,0.000000,,0.000000,USD,2355680.84,3626.60,30.55
